apps/ewallet/lib/ewallet/gates/blockchain/deposit_pooling_gate.ex

```diff
@@ -0,0 +1,145 @@
+# Copyright 2018-2019 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule EWallet.DepositPoolingGate do
+  @moduledoc """
+  Monitors deposit wallets and moves the funds into a pooled wallet when criteria are met.
+  """
+  require Logger
+  alias ActivityLogger.System
+  alias EWallet.{BlockchainHelper, TransactionRegistry, TransactionTracker}
+
+  alias EWalletDB.{
+    Token,
+    BlockchainWallet,
+    BlockchainDepositWalletBalance,
+    DepositTransaction,
+    Helpers.Preloader,
+    TransactionState
+  }
+
+  alias Keychain.Wallet
+
+  # TODO: get real gas price
+  @gas_price 20_000_000_000
+
+  def move_deposits_to_pooled_funds(blockchain_identifier) do
+    primary_token_address = BlockchainHelper.adapter().helper().default_token()[:address]
+    hot_wallet = BlockchainWallet.get_primary_hot_wallet(blockchain_identifier)
+
+    # We loop by token because it's likely we pool per specific token than a specific wallet.
+    # E.g. some tokens have more deposits than others, or the hot wallet may run out of one token
+    # more often than others. In this case, loop by token is O(n) while loop by wallet is O(n2).
+    pool_transactions =
+      blockchain_identifier
+      |> Token.all_blockchain()
+      |> Enum.flat_map(fn token ->
+        case token.blockchain_address do
+          ^primary_token_address ->
+            pool_token_deposits(token, hot_wallet, blockchain_identifier, @gas_price, 21_000)
```

Comment:
Don't want to put the amount (`21_000`) as module variable as well? So we can remember to update it in the `TODO`.